merlin/analysis/optimize.py

```diff
@@ -59,9 +65,25 @@ def _run_analysis(self, fragmentIndex):
                 self.parameters['preprocess_task'])
         codebook = self.get_codebook()
 
-        fovIndex = np.random.choice(list(self.dataSet.get_fovs()))
-        zIndex = np.random.choice(
-            list(range(len(self.dataSet.get_z_positions()))))
+        # Use random FOV and Z index if the user did not specify.
+        if 'fov_index' not in self.parameters:
+            fovIndex = np.random.choice(list(self.dataSet.get_fovs()))
+            zIndex = np.random.choice(
+                list(range(len(self.dataSet.get_z_positions()))))
+
+        # Otherwise use FOV and Z indices specified by the user.
+        else:
+            tmp = self.parameters['fov_index'][fragmentIndex]
+
+            # Two element list is FOV, z index.
+            if isinstance(tmp, list):
+                fovIndex = tmp[0]
+                zIndex = tmp[1]
+
+            # Otherwise this is the FOV index.
+            else:
+                fovIndex = tmp
+                zIndex = 0
```

Comment:
I might prefer only allowing the full specification of fov and zIndex are specified (such as [[0, 0], [1, 1], [2,2], [3, 3]]) and not the abbreviated specification of only fov. It is not clear that defaulting to zIndex 0 is always best. zIndex 0 can have artifacts from the surface and is likely the worst zIndex to optimize on.